docs/src/DeveloperDocumentation/styleguide.md

```diff
@@ -241,3 +217,189 @@ existing block.
 
  - Via the MathJax integration it is possible to use LaTeX code, and this is the
    preferred way to denote the mathematical symbols in the docstrings.
+
+
+## Printing in Oscar
+
+### The 2 + 1 print modes of Oscar
+Oscar has two user print modes `:details` and `:oneline` and one internal
+print mode `:supercompact`. The latter is for use during recursion,
+e.g. to print the `base_ring(X)` when in `:oneline` mode.
+It exists to make sure that `:oneline` stays compact and human readable.
+
+Top-level REPL printing of an object will use `:details` mode by default
+```julia
+julia> X
+:details
+```
+Inside nested structures, e.g. inside a `Vector`, the `:oneline` mode is used.
+```julia
+julia> [X,X]
+3-element Vector{TypeofX{T}}
+:oneline
+:oneline
+:oneline
+```
+
+##### An Example for the 2 + 1 print modes
+```
+# detailed
+General linear group of degree 24
+  over Finite field of degree 7 over GF(29)
+
+# oneline
+General linear group of degree 24 over GF(29^7)
+
+# supercompact
+General linear group
+```
+
+The print modes are specified as follows
+#### Detailed printing
+- the output must make sense as a standalone without context to non-specialists
+- the number of output lines should fit in the terminal
+- if the object is simple enough use only one line
+- use indentation and (usually) `:oneline` to print substructures
+#### One line printing
+- the output must print in one line
+- should make sense as a standalone without context
+- variable names/generators/relations should not be printed only their number.
+- Only the first word is capitalized e.g. `Polynomial ring`
+- one should use `:supercompact` for nested printing in compact
+- nested calls to `:oneline` (if you think them really necessary) should be at the end,
+  so that one can read sequentially. Calls to `:supercompact` can be anywhere.
+#### Super compact printing
+- no nested printing
+- a single term or a symbol/letter mimicking mathematical notation
+- should usually only depend on the type and not of the type parameters or of
+  the concrete instance - exceptions of this rule are possible e.g. for `GF(2)`
+- no nested printing. In particular variable names and `base_ring` must not be displayed.
```

Comment:
Nitpick: rules 1 and 4 are "no nested printing"

Reply:
Thanks.